### PR TITLE
bcm2835-codec: add support for V4L2_CID_MPEG_VIDEO_FORCE_KEY_FRAME

### PR DESCRIPTION
**Pull request by tmm1**
cc #3171 @6by9

### CODE REVIEW

**Comment by tmm1** [comment]
I tested this and it works as expected!

File: drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c [hunk 1]
```diff
@@ -2355,6 +2369,9 @@ static int bcm2835_codec_open(struct file *file)
 					 BIT(V4L2_MPEG_VIDEO_H264_PROFILE_MAIN) |
 					 BIT(V4L2_MPEG_VIDEO_H264_PROFILE_HIGH)),
 					V4L2_MPEG_VIDEO_H264_PROFILE_HIGH);
+		v4l2_ctrl_new_std(hdl, &bcm2835_codec_ctrl_ops,
```
Comment (6by9) [inline drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c:2372]: In adding an extra control, the size hint in the call ```v4l2_ctrl_handler_init(hdl, 6);``` above ought to be incremented by 1, to 7.

**Review by 6by9 — COMMENTED** [review]
Generally looks fine other than the one comment made below.

I'd like to do a quick test for myself, and also check that it is checkpatch clean, but otherwise that's my only comment.

File: drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c (hunk 1 above)
Comment (tmm1) [reply drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c:2372]: Fixed!

**Comment by tmm1** [comment]
Ran `checkpatch` and fixed this issue:

```
CHECK: Alignment should match open parenthesis
#2373: FILE: drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c:2373:
+		v4l2_ctrl_new_std(hdl, &bcm2835_codec_ctrl_ops,
+					V4L2_CID_MPEG_VIDEO_FORCE_KEY_FRAME,
```

**Review by 6by9 — APPROVED** [review]

**Comment by 6by9** [comment]
Thank you!
I'm good with these changes, but I'll let @popcornmix or @pelwell review them as well.

**Comment by tmm1** [comment]
Thanks!

How often are new kernels built for rpi-update?

**Comment by 6by9** [comment]
Depends on what changes have been made, but typically between 2 and 14 days. Most frequently it is with upstream kernel bumps.

**Pull request merged**